src/awkward/cppyy.py

```diff
@@ -0,0 +1,30 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+import awkward as ak
+
+_has_checked_version = False
+
+
+def register_and_check(array=None, cpp_type=None):
+    global _has_checked_version
+
+    try:
+        import cppyy
+    except ImportError as err:
+        raise ImportError(
+            """install the 'cppyy' package with:
+
+pip install cppyy
+
+or
+
+conda install -c conda-forge cppyy"""
```

Comment:
Users need to be warned. If they just try this and have ROOT installed, it would fail somehow, either in the installation phase or when trying to import it.

```suggestion
            """install the 'cppyy' package with:

pip install cppyy

or

conda install -c conda-forge cppyy

Note that this must be in a different venv or conda environment from ROOT, if you have installed ROOT.
"""
```